cmake/alpakaCommon.cmake

```diff
@@ -423,67 +423,64 @@ if(alpaka_ACC_GPU_CUDA_ENABLE)
 
             if(alpaka_FAST_MATH STREQUAL ON)
-                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:--use_fast_math>)
+                alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--use_fast_math>")
             endif()
 
             if(alpaka_FTZ STREQUAL ON)
-                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:--ftz=true>)
+                alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--ftz=true>")
             elseif(alpaka_FTZ STREQUAL OFF)
-                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:--ftz=false>)
+                alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--ftz=false>")
             endif()
 
             if(alpaka_CUDA_SHOW_REGISTER STREQUAL ON)
-                alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xptxas=-v>)
+                alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xptxas -v>")
             endif()
 
             if(alpaka_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR alpaka_ACC_CPU_B_SEQ_T_OMP2_ENABLE)
                 if(NOT MSVC)
-                    alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-fopenmp>)
+                    alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcompiler -fopenmp>")
 
                     # See https://github.com/alpaka-group/alpaka/issues/1755
                     if((${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang") AND
                        (${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 13))
                        message(STATUS "clang >= 13 detected. Force-setting OpenMP to version 4.5.")
-                       alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=-fopenmp-version=45>)
+                       alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcompiler -fopenmp-version=45>")
                     endif()
                 else()
-                    alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=/openmp>)
+                    alpaka_set_compiler_options(DEVICE target alpaka "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:-Xcompiler /openmp>")
                 endif()
             endif()
 
             # Always add warning/error numbers which can be used for suppressions
-            alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe=--display_error_number>)
-
-            # avoids warnings on host-device signatured, default constructors/destructors
-            alpaka_set_compiler_options(DEVICE target alpaka $<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe=--diag_suppress=esa_on_defaulted_function_ignored>)
```

Comment:
We don't have any `default`ed methods which are annotated with `ALPAKA_FN{_HOST}_ACC` so I don't see a reason to keep this warning.